src/filesystem/unix/SDL_sysfilesystem.c

```diff
@@ -332,4 +333,122 @@ SDL_GetPrefPath(const char *org, const char *app)
     return retval;
 }
 
+char *
+getpath_try_xdg(SDL_Folder folder)
+{
+    const char *param = NULL;
+    char cmd[32]; /* 32 bytes should be enough, increase if needed */
+    FILE *xdg_proc;
+    char buffer[4096]; /* Apparently PATH_MAX is unreliable */
+    size_t bytes_read;
+    char *retval;
+
+    /* According to `man xdg-user-dir`, the possible values are:
+        DESKTOP
+        DOWNLOAD
+        TEMPLATES
+        PUBLICSHARE
+        DOCUMENTS
+        MUSIC
+        PICTURES
+        VIDEOS
+    */
+    switch(folder)
+    {
+        case SDL_FOLDER_HOME:
+            return SDL_getenv("HOME");
+
+        case SDL_FOLDER_DESKTOP:
+            param = "DESKTOP";
+            break;
+
+        case SDL_FOLDER_DOCUMENTS:
+            param = "DOCUMENTS";
+            break;
+
+        case SDL_FOLDER_DOWNLOADS:
+            param = "DOWNLOADS";
+            break;
+
+        case SDL_FOLDER_MUSIC:
+            param = "MUSIC";
+            break;
+
+        case SDL_FOLDER_PICTURES:
+            param = "PICTURES";
+            break;
+
+        case SDL_FOLDER_VIDEOS:
+            param = "VIDEOS";
+            break;
+
+        case SDL_FOLDER_SCREENSHOTS:
+        default:
+            SDL_SetError("No corresponding XDG user directory");
+            return NULL;
+    }
+
+    /* param *should* to be set to something at this point, but just in case */
+    if (!param)
+    {
+        SDL_SetError("No corresponding XDG user directory");
+        return NULL;
+    }
+
+    if (snprintf(cmd, sizeof(cmd), "xdg-user-dir %s", param) >= sizeof(cmd))
+    {
+        /* If this happens, increase the size of `cmd` */
+        SDL_SetError("XDG command would be too long");
+        return NULL;
+    }
+
+    xdg_proc = popen(cmd, "r");
+    if (!xdg_proc)
+    {
+        SDL_SetError("Couldn't invoke xdg-user-dir: %s", strerror(errno));
+        return NULL;
+    }
+
+    bytes_read = fread(buffer, 1, sizeof(buffer), xdg_proc);
+
+    pclose(xdg_proc);
+
+    if (bytes_read > sizeof(buffer))
+    {
+        /* If this happens, increase the size of `buffer` */
+        SDL_SetError("Path too long");
+        return NULL;
+    }
+
+    /* The XDG command will finish with a newline; erase it */
+    buffer[bytes_read - 1] = '\0';
+
+    /* xdg-user-dir will return the home folder if a certain folder doesn't
+       exist; since SDL_FOLDER_HOME would've returned in the switch-case, if
+       xdg returned the home folder, then the folder doesn't exist */
+    if (!strcmp(buffer, SDL_getenv("HOME")))
+    {
+        SDL_SetError("XDG folder not found");
```

Comment:
Because setting one of these to be the same as `$HOME`, for example `XDG_DOCUMENTS_DIR=$HOME`, is documented to be how you disable a special folder, it would perhaps be more accurate to say "disabled by user" rather than "not found"?

Reply:
Is "not available" good terminology? It would cover both cases where the folder is disabled and cases where the folder has been omitted from the config file.